food-and-agriculture-organization-of-the-united-nations.csl

```diff
@@ -10,196 +10,368 @@
-    <group prefix=" ">
-      <text variable="publisher-place" suffix=","/>
-      <text variable="publisher" prefix=" " suffix="."/>
+  <macro name="date">
+    <!-- date format: year / undated / put "forthcoming" or else in "Date" field when needed.-->
+    <choose>
+      <if match="none" variable="issued">
+        <text term="no date"/>
+      </if>
+      <else>
+        <date date-parts="year" form="text" variable="issued"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher-and-place">
+    <!-- place / place, publisher -->
+    <group>
+      <text variable="publisher-place"/>
+      <choose>
+        <if variable="publisher">
+          <text variable="publisher" prefix=", "/>
+        </if>
+      </choose>
     </group>
   </macro>
   <macro name="edition">
-    <number font-style="italic" suffix=" " variable="edition" form="ordinal"/>
-    <text term="edition" font-style="italic"/>
+    <!-- Second edition -->
+    <number text-case="capitalize-first" variable="edition" form="long-ordinal"/>
+    <text term="edition" prefix=" "/>
+  </macro>
+  <macro name="Cite-and-URL">
+    <!-- [Cited day month year]. URL -->
+    <choose>
+      <if variable="URL">
+        <text term="cited" text-case="capitalize-first" prefix="["/>
+        <date form="text" variable="accessed" prefix=" " suffix="]."/>
+        <text variable="URL" prefix=" "/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="Available-at">
+    <!-- (also available at URL). -->
+    <choose>
+      <if variable="URL">
+        <text term="available at" text-case="lowercase" prefix="(also "/>
+        <text variable="URL" prefix=" " suffix=")."/>
+      </if>
+    </choose>
   </macro>
-  <macro name="Available-at-and-accessed">
-    <text term="available at" text-case="capitalize-first" suffix=": "/>
-    <text variable="URL" suffix=". "/>
-    <text term="accessed" text-case="capitalize-first" suffix=" "/>
-    <date form="text" variable="accessed" suffix="."/>
+  <macro name="number_of_pages-label">
+    <!-- 1 p. / x pp. -->
+    <choose>
+      <if is-numeric="number-of-pages">
+        <text variable="number-of-pages"/>
+        <label prefix=" " variable="number-of-pages" form="short"/>
+      </if>
+    </choose>
   </macro>
+  <!-- in-line citation: (author, date) -->
   <citation disambiguate-add-year-suffix="true" collapse="year">
     <sort>
       <key variable="author"/>
       <key variable="issued"/>
       <key variable="accessed"/>
     </sort>
     <layout prefix="(" suffix=")" delimiter="; ">
-      <group delimiter=", ">
-        <text macro="author-short"/>
-        <choose>
-          <if match="any" variable="issued">
-            <date date-parts="year" form="text" variable="issued"/>
-          </if>
-          <else>
-            <text value="undated"/>
-          </else>
-        </choose>
-        <text variable="locator"/>
-      </group>
+      <text macro="author-short" suffix=", "/>
+      <text macro="date"/>
```

Comment:
just to confirm -- there will never be page numbers or other locators given in the text?